resources/hvac.rb

```diff
@@ -2118,40 +2104,57 @@ def self.apply_dehumidifier(model, runner, energy_factor, water_removal_rate,
-        zone_hvac.setRatedWaterRemoval(UnitConversions.convert(water_removal_rate.to_f, "pint", "L"))
-      else
-        zone_hvac.setRatedWaterRemoval(Constants.small) # Autosize flag for HVACSizing measure
-      end
-      if energy_factor != Constants.Auto
-        zone_hvac.setRatedEnergyFactor(energy_factor.to_f)
-      else
-        zone_hvac.setRatedEnergyFactor(Constants.small) # Autosize flag for HVACSizing measure
-      end
-      if air_flow_rate != Constants.Auto
-        zone_hvac.setRatedAirFlowRate(UnitConversions.convert(air_flow_rate.to_f, "cfm", "m^3/s"))
-      else
-        zone_hvac.setRatedAirFlowRate(Constants.small) # Autosize flag for HVACSizing measure
-      end
-      zone_hvac.setMinimumDryBulbTemperatureforDehumidifierOperation(10)
-      zone_hvac.setMaximumDryBulbTemperatureforDehumidifierOperation(40)
+    w_coeff = [-1.162525707, 0.02271469, -0.000113208, 0.021110538, -0.0000693034, 0.000378843]
+    ef_coeff = [-1.902154518, 0.063466565, -0.000622839, 0.039540407, -0.000125637, -0.000176722]
+    pl_coeff = [0.90, 0.10, 0.0]
+    water_removal_curve = create_curve_biquadratic(model, w_coeff, "DXDH-WaterRemove-Cap-fT", -100, 100, -100, 100)
+    energy_factor_curve = create_curve_biquadratic(model, ef_coeff, "DXDH-EnergyFactor-fT", -100, 100, -100, 100)
+    part_load_frac_curve = create_curve_quadratic(model, pl_coeff, "DXDH-PLF-fPLR", 0, 1, 0.7, 1)
+
+    if energy_factor.nil?
+      # shift inputs tested under IEF test conditions to those under EF test conditions with performance curves
+      water_removal_rate_l = UnitConversions.convert(water_removal_rate, "pint", "L")
+      energy_factor, water_removal_rate = dehumidifier_ief_to_ef_inputs(w_coeff, ef_coeff, integrated_energy_factor, water_removal_rate_l)
+    end
+
+    humidistat = OpenStudio::Model::ZoneControlHumidistat.new(model)
+    humidistat.setName(obj_name + " #{control_zone.name} humidistat")
+    humidistat.setHumidifyingRelativeHumiditySetpointSchedule(relative_humidity_setpoint_sch)
+    humidistat.setDehumidifyingRelativeHumiditySetpointSchedule(relative_humidity_setpoint_sch)
+    control_zone.setZoneControlHumidistat(humidistat)
+
+    zone_hvac = OpenStudio::Model::ZoneHVACDehumidifierDX.new(model, water_removal_curve, energy_factor_curve, part_load_frac_curve)
+    zone_hvac.setName(obj_name + " #{control_zone.name} dx")
+    zone_hvac.setAvailabilitySchedule(model.alwaysOnDiscreteSchedule)
+    zone_hvac.setRatedWaterRemoval(water_removal_rate)
+    zone_hvac.setRatedEnergyFactor(energy_factor)
+    zone_hvac.setRatedAirFlowRate(UnitConversions.convert(air_flow_rate, "cfm", "m^3/s"))
+    zone_hvac.setMinimumDryBulbTemperatureforDehumidifierOperation(10)
+    zone_hvac.setMaximumDryBulbTemperatureforDehumidifierOperation(40)
+
+    zone_hvac.addToThermalZone(control_zone)
+  end
 
-      zone_hvac.addToThermalZone(control_zone)
-    end
+  def self.dehumidifier_ief_to_ef_inputs(w_coeff, ef_coeff, ief, water_removal_rate)
```

Comment:
Changing performance curve to be normalized to 1.0 at IEF test point is more complicated than directly applying IEF conditions to the existing curve and calculating the E+ rated inputs backwards. So I ended up with the latter approach. They should yield the same results if we want to keep the same performance curve shape. 
@jonwinkler can you take a quick look at the approach to make sure it's following what you suggested?